File: Data Structures & Algorithms/climbing-stairs/submission-0.cpp
class Solution {
public:
    int climbStairs(int n) {
        //n==1 -> 1
        //n==2 -> 1+1, 2 => 2
        //n==3 -> (n-1) + (n-2)*2
        int a = 1, b = 2, c;
        if(n==1){
            return a;
        }
        if(n==2){
            return b;
        }
        for(int i = 3; i<n+1; i++){
            c = b + a;
            a = b;
            b = c;
        }

        return c;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {

    private int[] getAns(int a, int b){
        int[] arr = new int[2];
        arr[0] = a;
        arr[1] = b;
        return arr;
    }

    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();

        for(int i = 0; i<nums.length; i++){
            int req = target - nums[i];
            if(map.containsKey(req)){
                return getAns(map.get(req), i);
            }
            else{
                map.put(nums[i], i);
            }
        }

        return getAns(-1, -1);
    }
}
